File: app/src/main/java/com/cliniva/enventory/viewholder/BrandHolder.java
package com.cliniva.enventory.viewholder;

import android.databinding.ViewDataBinding;
import android.support.annotation.NonNull;

import com.cliniva.enventory.adapter.base.BaseRecyclerClickListener;
import com.cliniva.enventory.adapter.base.BaseRecyclerViewHolder;
import com.cliniva.enventory.databinding.ItemBrandBinding;
import com.cliniva.enventory.model.Brand;

public class BrandHolder extends BaseRecyclerViewHolder<Brand, BaseRecyclerClickListener<Brand>> {

    private ItemBrandBinding binding;

    public BrandHolder(@NonNull ViewDataBinding itemView) {
        super(itemView);
        this.binding = (ItemBrandBinding) itemView;
    }

    @Override
    public void onBindView(Brand item, int position, BaseRecyclerClickListener<Brand> listener) {
        enableItemViewClick(item, listener);
        binding.setBrand(item);
    }
}

File: app/src/main/java/com/amirsons/inventory/recyclerview/viewholder/BrandHolder.kt
package com.amirsons.inventory.recyclerview.viewholder

import android.view.View
import com.amirsons.inventory.datamanager.model.BrandProduct
import com.amirsons.inventory.recyclerview.base.BaseRecyclerClickListener
import com.amirsons.inventory.recyclerview.base.BaseRecyclerViewHolder
import kotlinx.android.synthetic.main.item_brand.view.*

class BrandHolder(itemView: View) : BaseRecyclerViewHolder<BrandProduct, BaseRecyclerClickListener<BrandProduct>>(itemView) {

    override fun onBindView(item: BrandProduct, position: Int, listener: BaseRecyclerClickListener<BrandProduct>?) {

        itemView.tv_item_brand_name.text = item.brand

        // change the selected background
        itemView.tv_item_brand_name.isSelected = item.isSelected
        itemView.tv_item_brand_name.elevation = if (item.isSelected) 5f else 0f

        itemView.setOnClickListener {

            if (listener != null) {

                listener.onItemClickListener(item, position)

                // change the current selected background
                itemView.tv_item_brand_name.isSelected = true
                itemView.tv_item_brand_name.elevation = 5f
            }
        }
    }
}
